src/app/src/mainwindow.cpp

```diff
@@ -168,6 +168,8 @@ void MainWindow::Initialize() {
 
     //initial camera setup, need this to allocate buffers
     m_camera->SetupExp(m_expSettings);
+    // SetupExp will set this value, so need to update here
+    m_expSettings.bufferCount = m_camera->ctx->curExp->bufferCount;
```

Comment:
yeah, the way I originally wrote it so it would allocate at the start of each new acquisition but now that it doesn't do that it won't work. I don't really like having the state stored in two spots like this but it would probably take more work to fix that than it's worth

Reply:
one option is to have `SetupExp` either return the updated ExpSettings or pass in the value as a non-const reference and update `m_expSettings` in the `SetupExp` method itself

Reply:
I'll try updating the ref